projects/domain-events-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/engagement/EngagementResource.kt

```diff
@@ -11,7 +11,7 @@ import uk.gov.justice.digital.hmpps.service.PersonService
 @RestController
 @RequestMapping
 class EngagementResource(private val personService: PersonService) {
-    @PreAuthorize("hasRole('VIEW_PROBATION_CASE_ENGAGEMENT_CREATED')")
+    @PreAuthorize("hasAnyRole('VIEW_PROBATION_CASE_ENGAGEMENT_CREATED','PROBATION_API__DOMAIN_EVENTS__CASE_DETAIL')")
```

Comment:
Maybe a question for @mxco86 but should we potentially have separate roles per endpoint for domain event detail URLs?  Something like `PROBATION_API__DOMAIN_EVENTS__ENGAGEMENT` instead of `PROBATION_API__DOMAIN_EVENTS__CASE_DETAIL`.

Reply:
Yep agree - it would enable to control access based on the specific domain event

Reply:
changed to PROBATION_API__DOMAIN_EVENTS__ENGAGEMENT